flink-runtime/src/test/java/org/apache/flink/runtime/rest/RestServerEndpointITCase.java

```diff
@@ -661,12 +657,10 @@ protected void startInternal() {}
 				// Intentionally schedule the work on a different thread. This is to simulate
 				// handlers where the CompletableFuture is finished by the RPC framework.
 				return CompletableFuture.supplyAsync(() -> {
-					synchronized (LOCK) {
-						try {
-							LOCK.notifyAll();
-							LOCK.wait();
```

Comment:
Replaced with `CountDownLatch` because there can be spurious wakeups:

- https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait--
- https://stackoverflow.com/questions/1050592/do-spurious-wakeups-in-java-actually-happen